### DISCUSSION TITLE
607

### DISCUSSION THREAD

**Discussion by giscus** [jojoldu/blog-comments#926] [Comments]
# 607

NestJS 공식 문서에서는 아래와 같이 CLI로 서비스 실행을 소개하곤 하는데요. nest start --watch 물론 대부분의 백엔드 API 개발에 있어서는 nest start 를 쓸 일이 거의 없습니다. 대부분 테스트 코드로 단위 테스트 / E2E 테스트를 작성하면서 진행하다보니 서버 실행할 이유가 없기 때문입니다. 어떤 개발 환경에서도 터미널만 있으면 실행할 수 있으니 좋습니다만, 이럴 경우 WebStorm/IntelliJ 등 디버깅 기능이 강력한 IDE의 도움을 못받습니다. 개발에 있어서 디버거의 사용유무는 중요합니다. 실시간 데이터 변경 흐름 파악 특정 지점에서의 별도 로직 수행 Logger를 추가할 수 없는 외부 라이브러리에 대한 코드 추적 등등 디버거를 통해 좀 더 깊게 문제를 확인할 수..

https://jojoldu.tistory.com/607

**Comment by mystyle2006** [comment]
혹시 debug로 실행하실 때 안할 때와 비교하면 최초 빌드하는 속도가 느리지 않으신가요... 저는 이것 때문에 왠만하면 --watch 모드로 개발하다 디버깅이 필요할 때만 실행해서 개발합니다 ㅠ..
